### PR TITLE
[ao] qconfig.py fix public v private

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #88404
* #88400
* #88399
* #88398
* #88397
* #88396
* #88395
* #88394
* #88393
* #88392
* #88391
* #87885
* #87883
* #87521
* #87520
* #87519
* #87518
* #87517
* #87516
* __->__ #87515

Summary: made is_reuse_input_qconfig, _activation_is_memoryless,
_partial_wrapper_equals, _obs_or_fq_ctr_equals,
_add_module_to_qconfig_obs_ctr, _assert_valid_qconfig private

Test Plan: python test/test_public_bindings.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D40709280](https://our.internmc.facebook.com/intern/diff/D40709280)

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

### CODE REVIEW

File: test/allowlist_for_publicAPI.json [hunk 1]
```diff
@@ -622,7 +622,7 @@
     "OrderedDict"
   ],
   "torch.nn.qat.dynamic.modules.linear": [
-    "activation_is_memoryless"
+    "_activation_is_memoryless"
```
Comment (jerryzh168) [inline test/allowlist_for_publicAPI.json:625]: I think we can remove these after we made them private

Comment (anjali411) [reply test/allowlist_for_publicAPI.json:625]: yeah

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709280).

File: test/allowlist_for_publicAPI.json (hunk 1 above)
Comment (HDCharles) [reply test/allowlist_for_publicAPI.json:625]: i remove them at the end of the stack since there's a lot that can get removed

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709280).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709280).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709280).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D40709280).

File: test/allowlist_for_publicAPI.json (hunk 1 above)
Comment (HDCharles) [reply test/allowlist_for_publicAPI.json:626]: I remove them in the final PR

**Review by jcaip — APPROVED** [review]
lgtm

**Pull request closed without merging**